src/Stryker.Core/Stryker.Core/Options/ValidatedStrykerOptions.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Stryker.Core.Logging;
+using Stryker.Core.Mutators;
+using System.Collections.Generic;
+
+namespace Stryker.Core.Options
+{
+    public class ValidatedStrykerOptions
```

Comment:
What's the reasoning behind this class? I don't like classes like this as they shadow the original class (`StrykerOptions`).

Reply:
This is not a complete shadow class of `StrykerOptions`. StrykerOptions only has string values. In the `ValidatedStrykerOptions` this has been cast to int. Future properties could also have different types.

Reply:
I didn't say _copy_. If the name is the same, but type is different: it's still a shadow IMHO ;).

Couldn't we force the dotnet cli to interpret a value as an `int`? If not, i still think it should be done in the dotnet cli package, rather than the core Stryker package. Don't you agree?

Reply:
I didn't want to make the CLI responsible for casting values. So now the validator casts and validates all option values. If this is a huge issue I can change the casting to take place in the CLI. The need for the "Shadow" class would be gone that way.

Reply:
I think the CLI should be responsible for casting to the correct values. 

The reason why they are strings is because they are interpreted from the command line as strings. The Stryker core shouldn't be bothered with this detail, right? If you would want to reuse the code from a place where you would already have it as an int, you would need to cast it back to string. Doesn't make sense.

Validating that the int is in the expected range however _is_ a responsibility of the Stryker core.